atoma-output-manager/src/lib.rs

```diff
@@ -22,27 +27,59 @@ mod gateway;
 pub struct AtomaOutputManager {
     /// Firebase's output manager instance.
     firebase_output_manager: FirebaseOutputManager,
+    /// IPFS's join handle related to the IPFS output manager background task.
+    ipfs_join_handle: Option<JoinHandle<Result<(), AtomaOutputManagerError>>>,
+    /// Request sender to the IPFS manager background task.
+    ipfs_request_tx: mpsc::UnboundedSender<(AtomaOutputMetadata, Vec<u8>)>,
     /// Gateway's output manager.
     gateway_output_manager: GatewayOutputManager,
     /// A mpsc receiver that receives tuples of `AtomaOutputMetadata` and
     /// the actual AI generated output, in JSON format.
-    output_manager_rx: mpsc::Receiver<(AtomaOutputMetadata, String)>,
+    output_manager_rx: mpsc::Receiver<(AtomaOutputMetadata, Vec<u8>)>,
 }
 
 impl AtomaOutputManager {
     /// Constructor
     pub async fn new<P: AsRef<Path>>(
         config_file_path: P,
-        output_manager_rx: mpsc::Receiver<(AtomaOutputMetadata, String)>,
+        output_manager_rx: mpsc::Receiver<(AtomaOutputMetadata, Vec<u8>)>,
         firebase: Firebase,
     ) -> Result<Self, AtomaOutputManagerError> {
         let config = AtomaOutputManagerConfig::from_file_path(config_file_path);
-        let firebase_output_manager =
-            FirebaseOutputManager::new(firebase);
+        let firebase_output_manager = FirebaseOutputManager::new(firebase);
         let gateway_output_manager =
             GatewayOutputManager::new(&config.gateway_api_key, &config.gateway_bearer_token);
+
+        info!("Building IPFS client...");
+        let client = IpfsClient::default();
+        let (ipfs_request_tx, ipfs_request_rx) = mpsc::unbounded_channel();
+
+        let ipfs_join_handle = match client.version().await {
+            Ok(version) => {
+                info!(
+                    "IPFS client built successfully, with version = {:?}",
+                    version
+                );
+                let ipfs_join_handle = tokio::spawn(async move {
+                    let mut ipfs_output_manager =
+                        IpfsOutputManager::new(client, ipfs_request_rx).await?;
+                    ipfs_output_manager.run().await?;
+                    Ok(())
+                });
+                Some(ipfs_join_handle)
+            }
+            Err(e) => {
+                error!(
+                    "Failed to obtain IPFS client's version: {e}, most likely IPFS daemon is not running in the background. To start it, run `$ ipfs daemon`"
```

Comment:
This would be nice to have windows explainer as well, but I suppose not many people will run windows

Reply:
It's the same command